src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/ITemporaryStreamStorageExtensions.cs

```diff
@@ -2,29 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using System.IO;
 
 namespace Microsoft.CodeAnalysis.Host
 {
     internal static class ITemporaryStreamStorageExtensions
     {
-        public static void WriteString(this ITemporaryStreamStorage storage, string value)
+        public static void WriteAllLines(this ITemporaryStreamStorage storage, IEnumerable<string> values)
         {
             using var stream = SerializableBytes.CreateWritableStream();
             using var writer = new StreamWriter(stream);
 
-            writer.Write(value);
+            foreach (var value in values)
+            {
+                writer.Write(value);
+            }
+
             writer.Flush();
             stream.Position = 0;
 
             storage.WriteStream(stream);
         }
 
-        public static string ReadString(this ITemporaryStreamStorage storage)
+        public static IEnumerable<string> ReadLines(this ITemporaryStreamStorage storage)
         {
             using var stream = storage.ReadStream();
             using var reader = new StreamReader(stream);
-            return reader.ReadToEnd();
+
+            string line;
+            while ((line = reader.ReadLine()) != null)
+            {
+                yield return line;
```

Comment:
would prefer an immutable array.  the currently approach is lazy, but allows for multiple reads that go back to the storage.